edx/edx-enterprise
# [ENT-587] More gracefully handle errors from hitting the Ecommerce API (and others)

**Description:** This PR moves error handling logic for our external API clients into the clients themselves, rather than having try/except blocks all over the codebase.

**JIRA:** [ENT-587](https://openedx.atlassian.net/browse/ENT-587)

**Merge deadline:** Before the end of the week.

**Testing instructions:**

1. Since this is just a refactor and doesn't introduce any new functionality (but does add test coverage that was lost due to moving logic around), you can either rely on the tests or just give the course enrollment landing page a spin; see if the API calls are working properly and that in exceptional cases like missing certain data, we get the same behavior as last time.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** I do wonder if the defaults I return in some cases are appropriate. So please do a double-check there.
